go/worker/client/worker.go

```diff
@@ -124,8 +129,19 @@ func (w *Worker) registerRuntime(commonNode *committeeCommon.Node) error {
 		"runtime_id", id,
 	)
 
+	// When a node is a client node and it has an entity configured, we register it with the
+	// observer role as this may be needed for confidential runtimes.
+	var rp registration.RoleProvider
+	if config.GlobalConfig.Mode == config.ModeClient {
+		var err error
+		rp, err = w.registration.NewRuntimeRoleProvider(node.RoleObserver, id)
```

Comment:
Should `rp.SetAvailable(commonNode.RegisterNodeRuntime)` be moved here?

Reply:
Done.